test/cases/ostree-vsphere.sh

```diff
@@ -0,0 +1,787 @@
+name = "vmdk"
+description = "A rhel-edge vmdk image"
+version = "0.0.1"
+modules = []
+groups = []
+
+[[customizations.user]]
+name = "admin"
+description = "Administrator account"
+password = "\$6\$GRmb7S0p8vsYmXzH\$o0E020S.9JQGaHkszoog4ha4AQVs3sk8q0DvLjSMxoxHBKnB2FBXGQ/OkwZQfW/76ktHd0NX5nls2LPxPuUdl."
+key = "${SSH_KEY_PUB}"
+home = "/home/admin/"
+groups = ["wheel"]
+
+[customizations.ignition.firstboot]
+url = "${IGNITION_SERVER_URL}/config.ign"
+EOF
+
+greenprint "📄 vmdk blueprint"
+cat "$BLUEPRINT_FILE"
+
+# Prepare the blueprint for the compose.
+greenprint "📋 Preparing installer blueprint"
+sudo composer-cli blueprints push "$BLUEPRINT_FILE"
+sudo composer-cli blueprints depsolve vmdk
+
+# Build simplified installer iso image.
+build_image vmdk "${VSPHERE_IMAGE_TYPE}" "${PROD_REPO_URL}/"
+
+# Download the image
+greenprint "📥 Downloading the vmdk image"
+sudo composer-cli compose image "${COMPOSE_ID}" > /dev/null
+ISO_FILENAME="${COMPOSE_ID}-${VSPHERE_FILENAME}"
+sudo chmod 644 "${ISO_FILENAME}"
+
+# Clean compose and blueprints.
+greenprint "🧹 Clean up vmdk blueprint and compose"
+sudo composer-cli compose delete "${COMPOSE_ID}" > /dev/null
+sudo composer-cli blueprints delete vmdk > /dev/null
+
+##################################################################
+##
+## Upload image to datastore
+##
+##################################################################
+export GOVC_URL="${GOVC_URL}"
+export GOVC_USERNAME="${VSPHERE_USERNAME}"
+export GOVC_PASSWORD="${VSPHERE_PASSWORD}"
+export GOVC_INSECURE="true"
+
+greenprint "📋 Uploading vmdk image to vsphere datacenter 7.0"
+govc import.vmdk -dc="Datacenter7.0-AMD" -ds="datastore-21" -pool="/Datacenter7.0-AMD/host/Cluster7.0-AMD/Resources" "${ISO_FILENAME}"
+greenprint "📋 Copying vmdk image to vsphere datacenter 6.7"
+govc datastore.mkdir -dc="Datacenter6.7" -ds="datastore-225" "${COMPOSE_ID}-image"
+govc datastore.cp -dc="Datacenter7.0-AMD" -ds="datastore-21" -dc-target="Datacenter6.7" -ds-target="datastore-225" "${COMPOSE_ID}-image/${ISO_FILENAME}" "${COMPOSE_ID}-image/${ISO_FILENAME}"
+
+##################################################################
+##
+## Create vm on datacenter7.0-amd and test it
+##
+##################################################################
+# Create vm with vmdk image
+greenprint "📋 Create vm in vsphere datacenter 7.0-AMD"
+DC70_VSPHERE_VM_NAME="${COMPOSE_ID}-70"
+govc vm.create -dc="Datacenter7.0-AMD" -ds="datastore-21" -pool="/Datacenter7.0-AMD/host/Cluster7.0-AMD/Resources" \
+    -net="VM Network" -net.adapter=vmxnet3 -disk.controller=pvscsi -on=false -c=2 -m=4096 \
+    -g="${GUEST_ID_DC70}" -firmware=efi "${DC70_VSPHERE_VM_NAME}"
+govc vm.disk.attach -ds="datastore-21" -dc="Datacenter7.0-AMD" -vm "${DC70_VSPHERE_VM_NAME}" \
+    -link=false -disk="${COMPOSE_ID}-image/${ISO_FILENAME}"
+govc vm.power -on -dc="Datacenter7.0-AMD" "${DC70_VSPHERE_VM_NAME}"
+DC70_GUEST_ADDRESS=$(govc vm.ip -dc=Datacenter7.0-AMD -wait=10m "${DC70_VSPHERE_VM_NAME}")
```

Comment:
Could you please add `-v4` in this command to avoid IPv6 address when it becomes available.

Reply:
Fixed.